eng/tools/azure-sdk-tools/azpysdk/sphinx.py

```diff
@@ -0,0 +1,335 @@
+
+    output = ""
+    if readme_ext == ".md":
+        with open(readme_location, "r") as file:
+            output = file.read()
+    else:
+        logger.error(
+            "{} is not a valid readme type. Expecting RST or MD.".format(
+                readme_location
+            )
+        )
+
+    output += RST_EXTENSION_FOR_INDEX.format(package_rst)
+
+    return output
+
+def create_index(doc_folder: str, source_location: str, namespace: str) -> None:
+    index_content = ""
+
+    package_rst = "{}.rst".format(namespace)
+    content_destination = os.path.join(doc_folder, "index.md")
+
+    if not os.path.exists(doc_folder):
+        os.mkdir(doc_folder)
+
+    # grep all content
+    markdown_readmes = glob.glob(os.path.join(source_location, "README.md"))
+
+    # if markdown, take that, otherwise rst
+    if markdown_readmes:
+        index_content = create_index_file(markdown_readmes[0], package_rst)
+    else:
+        logger.warning("No readmes detected for this namespace {}".format(namespace))
+        index_content = RST_EXTENSION_FOR_INDEX.format(package_rst)
+
+    # write index
+    with open(content_destination, "w+", encoding='utf-8') as f:
+        f.write(index_content)
+
+def write_version(site_folder: str, version: str) -> None:
+    if not os.path.isdir(site_folder):
+        os.mkdir(site_folder)
+
+    with open(os.path.join(site_folder, "version.txt"), "w") as f:
+        f.write(version)
+
+# apidoc helper functions
+def is_mgmt_package(pkg_name: str) -> bool:
+    return pkg_name != "azure-mgmt-core" and ("mgmt" in pkg_name or "cognitiveservices" in pkg_name)
+
+def copy_existing_docs(source: str, target: str) -> None:
+    for file in os.listdir(source):
+        logger.info("Copying {}".format(file))
+        shutil.copy(os.path.join(source, file), target)
+
+def mgmt_apidoc(output_dir: str, target_folder: str, executable: str) -> int:
+    command_array = [
+        executable,
+        generate_mgmt_script,
+        "-p",
+        target_folder,
+        "-o",
+        output_dir,
+        "--verbose",
+        ]
+
+    try:
+        logger.info("Command to generate management sphinx sources: {}".format(command_array))
+
+        check_call(
+            command_array
+        )
+    except CalledProcessError as e:
+        logger.error(
+            "script failed for path {} exited with error {}".format(
+                output_dir, e.returncode
+            )
+        )
+        return 1
+    return 0
+
+def sphinx_apidoc(output_dir: str, target_dir: str, namespace: str) -> int:
+    working_doc_folder = os.path.join(output_dir, "doc")
+    command_array = [
+            "sphinx-apidoc",
+            "--no-toc",
+            "--module-first",
+            "-o",
+            os.path.join(output_dir, "docgen"),   # This is the output folder
+            os.path.join(target_dir, ""),         # This is the input folder
+            os.path.join(target_dir, "test*"),    # Starting here this, is excluded
```

Comment:
Nit: please fix my awkward commenting text 😆 

`starting with this argument the rest are "exclude" directory arguments` or something.